From 75e3fba1661c45621b1d35903a8aaf2478f9a009 Mon Sep 17 00:00:00 2001
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Tue, 19 Dec 2017 16:50:34 -0500
Subject: [PATCH] STX: Stop processing memory stats if balloon fails

We were seeing deadlock on the second query because the VM
had already been destroyed.  In that case, the first query
fails but unfortunately we ignore the return code and charge
ahead anyway and do the second query which then hangs.  We
now respect the return code if it indicates failure on the
first query.

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
[ Trimmed the shortlog ]
Signed-off-by: Thales Elero Cervi <thaleselero.cervi@windriver.com>
[ Fixed goto cleanup while porting patch to libvirt v8.0.0-1 ]
Signed-off-by: Murillo Arantes <murillo.arantes@windriver.com>
[ Ported to libvirt v11.3.0-3+deb13u2 ]
Signed-off-by: Murillo Arantes <murillo.arantes@windriver.com>
---
 src/qemu/qemu_monitor_json.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/qemu/qemu_monitor_json.c b/src/qemu/qemu_monitor_json.c
index dc2eaace9..0d7eeef7a 100644
--- a/src/qemu/qemu_monitor_json.c
+++ b/src/qemu/qemu_monitor_json.c
@@ -1983,6 +1983,8 @@ qemuMonitorJSONGetMemoryStats(qemuMonitor *mon,
     int got = 0;
 
     ret = qemuMonitorJSONGetBalloonInfo(mon, &mem);
+    if (ret < 0)
+        return got;
     if (ret == 1 && (got < nr_stats)) {
         stats[got].tag = VIR_DOMAIN_MEMORY_STAT_ACTUAL_BALLOON;
         stats[got].val = mem;
-- 
2.34.1

